Lib/colorsys.py

```diff
@@ -130,15 +130,12 @@ def rgb_to_hsv(r, g, b):
     if minc == maxc:
```

Comment:
Does the function become faster for greyscale values if `rangec = (maxc-minc)` is moved after the if check?

Reply:
surprisingly, no
Old: 6.133919600164518
New: 6.310812799958512

Reply:
Yes:

```none
Old: 4.684144808999999
New: 4.411844383999999
```

Reply:
Strange, could you try running this?

```python
import timeit

def rgb_to_hsv_old(r, g, b):
    maxc = max(r, g, b)
    minc = min(r, g, b)
    rangec = (maxc-minc)
    v = maxc
    if minc == maxc:
        return 0.0, 0.0, v
    s = rangec / maxc
    if r == maxc:
        h = (g-b) / rangec
    elif g == maxc:
        h = 2.0 + (b-r) / rangec
    else:
        h = 4.0 + (r-g) / rangec
    h = (h/6.0) % 1.0
    return h, s, v

def rgb_to_hsv_new(r, g, b):
    maxc = max(r, g, b)
    minc = min(r, g, b)
    v = maxc
    if minc == maxc:
        return 0.0, 0.0, v
    rangec = (maxc-minc)
    s = rangec / maxc
    if r == maxc:
        h = (g-b) / rangec
    elif g == maxc:
        h = 2.0 + (b-r) / rangec
    else:
        h = 4.0 + (r-g) / rangec
    h = (h/6.0) % 1.0
    return h, s, v

print('Old:',timeit.timeit(lambda: rgb_to_hsv_old(9,9,9),number=10_000_000))
print('New:',timeit.timeit(lambda: rgb_to_hsv_new(9,9,9),number=10_000_000))
```

Reply:
Old: 4.841873000143096
New: 4.7804170998279005

The arguments to `rgb_to_hsv` are floats in the range [0,1] though, not integers.

Reply:
Yeah my bad, try with 0.5 then.

Reply:
Old: 4.947571000084281
New: 4.570939600002021

though with (0.2,0.5,0.7) I get

Old: 6.850879299920052
New: 6.9389673001132905

so it is faster when the numbers are the same, but slower when they aren't

Reply:
No, it's just random:

```none
Old: 5.283142290998512
New: 5.288310900999932
```